### ISSUE TITLE
sendDeliveryConfirmation notifications not getting received by original sender

### ISSUE THREAD

**Issue by matanvr** [magnetsystems/message-ios#74]
I am experiencing this issue in Magnet 2.1 - not sure if its fixed in a later commit.

Here's my code:

the receiver of a message calls  [message sendDeliveryConfirmation];

```


-(void) viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];

    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(didReceiveDeliveryConfirmation:)
                                                 name:MMXDidReceiveDeliveryConfirmationNotification
                                               object:nil];

    
    }


- (void) didReceiveDeliveryConfirmation:(NSNotification *)notification {
    MMUser *recipient = notification.userInfo[MMXRecipientKey];
    NSString *messageID = notification.userInfo[MMXMessageIDKey];
}



```

**Comment by priteshshah1983** [comment]
@matanvr It looks like the issue got cut-off. Can you please post it again?

**Comment by priteshshah1983** [comment]
Nevermind, I understand the issue. Let me take a look.

**Comment by matanvr** [comment]
Fixed the formatting

**Comment by priteshshah1983** [comment]
I just verified it works with `v2.3`

```
pod 'MagnetMax', '~> 2.3'
```

`v2.3` has API compatibility with `v2.1`. Are there reasons why you can't upgrade?

**Comment by matanvr** [comment]
Thanks for the prompt response Pritesh, I just got a chance to try it with the latest release (2.4). I'm still not receiving the notification on the sender.

**Comment by priteshshah1983** [comment]
Is the sender able to receive other messages? Can you ensure that you have `[MMX start];` in the login success block like below?
```objective-c
[MMUser login:credential success:^{
        
    // Indicate that you are ready to receive messages now!
    [MMX start];
        
} failure:^(NSError * error) {
    NSLog(@"[ERROR]: Failed to login: %@", error.localizedDescription);
}];
```

If this doesn't work let me know, I will post a sample working project on GitHub.

**Comment by priteshshah1983** [comment]
@matanvr Were you able to get this to work?

**Comment by matanvr** [comment]
Hey Pritesh, I just got around to trying your code example. It doesn't work
when I try to send a delivery confirmation to myself. Does delivery
confirmation matter require different username/userid's?

**Comment by priteshshah1983** [comment]
Hi @matanvr,
We tested this ourselves and it is working.To summarize, you need the following:

```
- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    
    // Indicate that you are ready to receive messages now!
    [MMX start];
    
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(didReceiveMessage:)
                                                 name:MMXDidReceiveMessageNotification object:nil];
    
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(didReceiveDeliveryConfirmation:)
                                                 name:MMXDidReceiveDeliveryConfirmationNotification
                                               object:nil];
}

- (void)didReceiveDeliveryConfirmation:(NSNotification *)notification {
    MMUser *recipient = notification.userInfo[MMXRecipientKey];
    NSString *messageID = notification.userInfo[MMXMessageIDKey];
    NSLog(@"Received delivery confirmation");
}

- (void)didReceiveMessage:(NSNotification *)notification {
    MMXMessage *message = notification.userInfo[MMXMessageKey];
    // Send delivery confirmation
    [message sendDeliveryConfirmation];
}
```

Are you missing anything from the above?

**Comment by priteshshah1983** [comment]
@matanvr Are you trying `sendDeliveryConfirmation` with channels? I assume this is the case. I found a bug and believe I have a fix in https://github.com/magnetsystems/message-ios/commit/63312670df27f06e41718fe96f5ec0b606786cee I will test more tomorrow and push a new version of the pod if all goes well.

**Comment by matanvr** [comment]
That is correct I am also seeing it on channels.

Thanks

**Comment by priteshshah1983** [comment]
@matanvr Thanks for your patience!

This issue is fixed in `v2.5.1`:

```
pod 'MagnetMax', '~> 2.5'
```

**Comment by priteshshah1983** [comment]
@matanvr Were you able to try this?

**Comment by matanvr** [comment]
I've been out of the country for some time. Will try it out soon. Thanks
for the fix!

**Comment by matanvr** [comment]
@priteshshah1983  I just tried again and it's not working for me. Any tips on how to debug what the problem could possibly be? My code is exactly the same as the example you sent. 

To be more specific what I am seeing in my code is the sendDeliveryConfirmation call being made on the message. My callbacks are not getting called though.

**Comment by priteshshah1983** [comment]
Hi @matanvr,
We found and fixed one more issue in `v2.5.3` of the MMX pod.

You should continue to use `pod 'MagnetMax', '~> 2.5'` but you need to reinstall the pod.

Let me know how it goes!
